src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -182,7 +183,19 @@ private int RunTests(TestPackage package, TestFilter filter)
                 {
                     throw new NUnitEngineException($"Error encountered in resolving output specification: {spec}", ex);
                 }
-
+                
+                try
+                {
+                    Directory.CreateDirectory(outputDirectory);
+                }
+                catch (SystemException ex)
+                {
+                    throw new NUnitEngineException(
```

Comment:
IF we do want to wrap the exception, an `NUnitEngineException` seems inappropriate. This exception has nothing to do with the engine. The runner is trying to create the directory itself and gets an exception from .NET. Wrapping it as an engine exception is potentially misleading.

Reviewing the console runner code, I can see at least one other place where `NUnitEngineException` is used in this way. It's directly after this code and may be the reason why it was used here.

Originally, `NUnitEngineException` was an exception thrown __only__ by the engine. It provided a way to clarify where an exception originated. Later, it came to be used in this file to wrap __other__ kinds of exceptions thrown by the engine. IMO, that's unfortunate, since it means that other runners have to figure out for themselves whether or how to wrap such exceptions. However, that's probably water under the bridge at that point.

I do, however, think it would be a good idea to stop using `NUnitEngineException` for errors that have nothing to do with the engine.

Reply:
That makes sense. I definitely used `NUnitEngineException` because it was used in the code directly below. As i said above in https://github.com/nunit/nunit-console/pull/539#discussion_r253116045 , I personally like the idea of wrapping the error with a high-level comment. After taking a look at the errors that can happen, I can think of three solutions:
1. Don't wrap the exceptions and let the original exception just bubble up  
2. Wrap it in a`SystemException` and throw that instead
3. Wrap it in a`OptionException` and throw that instead

Does wrapping it in an `OptionException` make sense? From what I can tell, it's thrown when the user incorrectly inputs options in the command line. This exception seems like it fits for this concern.

Reply:
`OptionException` belongs to the underlying Mono.Options software that we use to parse the command-line. IMO it would make sense to use it if the command-line parser were detecting the error. However, the parser is pretty much unaware of what the string arguments to options mean or how they are used by the application. Also, the CreateDirectory error can only be detected by trying to create the directory, which is definitely not the parser's job.

The other options seem valid to me. Another wrapper possibility is `InvalidOperationException`.  We would still have to ensure that the console displayed both the general wrapping message and the specific reason. Otherwise all the user knows is that some kind of problem happened when creating a directory.

Personally, for software that is used by developers, I have no problem with just throwing the original exception, but there's some logic to wrapping as well, so long as we are consistent. 

@nunit/engine-team Does somebody want to weigh in on this?